pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
+	}
+
+	clusterRoleDict := make(map[string][]string)
+	//map of Name to Roles
+	for _, e := range apiResponseMCDistributedPeers.Entry {
+		for _, s := range e.Content.ClusterLabel {
+			clusterRoleDict[e.Name] = append(clusterRoleDict[e.Name], s)
+		}
+	}
+	//TODO: check different labels here
+	clusterRoleDictToDict := make(map[string][]string)
+	for key, value := range clusterRoleDict {
+		for _, val := range value {
+			clusterRoleDictToDict[val] = append(clusterRoleDictToDict[val], key)
+		}
+	}
+
+	clusterRoleDictToDictString := make(map[string]string)
+	for key, value := range clusterRoleDictToDict {
+		for _, val := range value {
+			clusterRoleDictToDictString[key] = clusterRoleDictToDictString[key] + "&member=" + val
+		}
+	}
+
+	for key, value := range clusterRoleDictToDictString {
+		if key == "" {
+			break
+		} else {
+			err = c.UpdateDMCClusteringLabelGroup(key, value)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	apiResponseMCAssetTableBuild, err := c.GetMonitoringconsoleAssetTable()
+	if err != nil {
+		return err
+	}
+	err = c.PostMonitoringConsoleAssetTable(apiResponseMCAssetTableBuild)
+	if err != nil {
+		return err
+	}
+	UISettingsObject, err := c.GetMonitoringConsoleUISettings()
+	if err != nil {
+		return err
+	}
+	err = c.UpdateLookupUISettings(configuredPeers, UISettingsObject)
+	if err != nil {
+		return err
+	}
+	err = c.UpdateMonitoringConsoleApp()
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//GetMonitoringconsoleServerRoles to retrive server roles of the local host or SplunkMonitoringConsole
+func (c *SplunkClient) GetMonitoringconsoleServerRoles() (*MCServerRolesInfo, error) {
+	apiResponseServerRoles := struct {
+		Entry []struct {
+			Content MCServerRolesInfo `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/services/server/info/server-info"
+	err := c.Get(path, &apiResponseServerRoles)
+	if err != nil {
+		return nil, err
+	}
+	if len(apiResponseServerRoles.Entry) < 1 {
+		return nil, err
+	}
+	return &apiResponseServerRoles.Entry[0].Content, nil
+}
+
+//UpdateDMCGroups dmc* groups with new members
+func (c *SplunkClient) UpdateDMCGroups(dmcGroupName string, groupMembers string) error {
+	endpoint := fmt.Sprintf("%s/services/search/distributed/groups/%s/edit", c.ManagementURI, dmcGroupName)
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(groupMembers))
+	err = c.Do(request, "200, 201, 409", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//UpdateDMCClusteringLabelGroup update respective clustering group
+func (c *SplunkClient) UpdateDMCClusteringLabelGroup(groupName string, groupMembers string) error {
+	endpoint := fmt.Sprintf("%s/services/search/distributed/groups/dmc_indexerclustergroup_%s/edit", c.ManagementURI, groupName)
+	reqBodyClusterGroup := groupMembers + "&default=false"
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(reqBodyClusterGroup))
+	err = c.Do(request, "200, 201, 409", nil)
+	if err != nil {
+		return err
```

Comment:
Info request: Now that you are moving the MC logic to a later stage for the associated CR phase to be ready. Having that clarity, do you see any risk for the CR deployment if we keep falling for any of these REST call failures for MC?